compiler-core/src/language_server/engine.rs

```diff
@@ -674,6 +677,46 @@ fn code_action_unused_imports(
         .push_to(actions);
 }
 
+fn code_action_import_module(
+    module: &Module,
+    params: &lsp::CodeActionParams,
+    actions: &mut Vec<CodeAction>,
+) {
+    let uri = &params.text_document.uri;
+    let suggestions = &module.ast.type_info.import_suggestions;
+
+    if suggestions.is_empty() {
+        return;
+    }
+
+    let line_numbers = LineNumbers::new(&module.code);
+    let (first_import_pos, first_is_import) = first_import_in_module(module, &line_numbers);
+    let after_import_newlines = add_newlines_after_import(
+        first_import_pos,
+        first_is_import,
+        &line_numbers,
+        &module.code,
+    );
+
+    for suggestion in suggestions {
```

Comment:
This may produce an overwhelming amount of code actions. Instead let's have one code action and then it presents a menu of options, like Rust Analyzer does

Reply:
Looking into Rust Analyzer, it seems that this behaviour is not part of the Language Server Protocol. Instead, they implement custom code action handling on the client. Should I try and figure that out? It would need to be implemented separately for each editor we support, as far as I can tell.